scripts/DCOSWindowsAgentSetup.ps1

```diff
@@ -164,7 +164,10 @@ function Install-AdminRouterAgent {
 }
 
 function Install-DiagnosticsAgent {
-    & "$SCRIPTS_DIR\scripts\diagnostics-agent-setup.ps1" -DiagnosticsWindowsBinariesURL $DIAGNOSTICS_BINARIES_URL
+    Param(
+        [bool]$IncludeMatricsService
+    )
+    & "$SCRIPTS_DIR\scripts\diagnostics-agent-setup.ps1" -IncludeMetricsToMonitoredSericeList $IncludeMatricsService
```

Comment:
Just something that stroke my mind right now to simplify the code a little bit more.
Sorry for not including it in the initial review.

Instead of creating the below variable ` $IsMatricsServiceInstalled` and pass it as the parameter `$IncludeMatricsService`, we could simply call `Get-MesosFlags` again and create the boolean here.

Have something like this:

```
$mesosFlags = Get-MesosFlags
$includeMetricsService = -not $mesosFlags.authenticate_agents
& "$SCRIPTS_DIR\scripts\diagnostics-agent-setup.ps1" -IncludeMetricsToMonitoredSericeList $includeMetricsService
```

And simply remove the parameter `$IncludeMatricsService` and the variable `$IsMatricsServiceInstalled` from below.

By the way, there is no concern calling `Get-MesosFlags` since it's already called before and it will return instantly.
Only the first time you call it, if the DC/OS API is not ready, it will keep on retrying.

Reply:
For this one, I think adding installation option parameter to Install-DiagnosticsAgent is more logical and efficient than having to call Get-MesosFlags in two places.

Reply:
True that as well, it just made the code look cleaner.

I agree, let's leave it like this.